### ISSUE TITLE
[diagnostics] Produce backtraces automatically when an error is an anomaly.

### ISSUE THREAD

**Issue by ejgallego** [ejgallego/coq-lsp#153]

**Comment by Alizter** [comment]
We basically want to:
1. Encounter an anomalous sentence.
2. Notice it is an anomaly.
3. Rerun it with a `Debug "backtrace".` prepended.

This should be similar to the implementation of auto-admitted Qeds right?

**Comment by ejgallego** [comment]
> We basically want to:
> 
> 1. Encounter an anomalous sentence.
> 2. Notice it is an anomaly.
> 3. Rerun it with a `Debug "backtrace".` prepended.

Yes.

> This should be similar to the implementation of auto-admitted Qeds right?

Good question. I was not thinking of using the error recovery system for this, but actually hooking at a lower level (like in `Coq.Protect`) so for `fleche` the process is transparent, i.e. it already gets the right exn, but what you propose could also make sense and maybe be simpler?

I guess we need to try.

**Comment by ejgallego** [comment]
Yeah, after thinking a bit more about this, let's try to do this in `Coq.Protect`. However we'll need to strengthen the invariant of callers as to ensure they are fully pure.
